prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/channel/ChannelAddCommand.java

```diff
@@ -43,36 +45,55 @@ public class ChannelAddCommand extends AbstractCommand {
+        private List<String> repositoryDefs;
+    }
+    @CommandLine.Option(names = CliConstants.CHANNEL)
+    private String channelLocation;
 
     @CommandLine.Option(names = CliConstants.DIR)
     private Optional<Path> directory;
 
-    @CommandLine.Option(names = CliConstants.REPOSITORIES, split = ",", paramLabel = CliConstants.REPO_URL, required = true)
-    private List<String> repositoryDefs;
-
     public ChannelAddCommand(CliConsole console, ActionFactory actionFactory) {
         super(console, actionFactory);
     }
 
     @Override
     public Integer call() throws Exception {
         Path installationDirectory = determineInstallationDirectory(directory);
+        List<Channel> channels = null;
+        if (channelLocation != null && !channelLocation.isEmpty()) {
+            String yaml = Files.readString(Path.of(channelLocation));
+            channels = ChannelMapper.fromString(yaml);
+            Channel channel = channels.get(0);
+            channels.set(0, new Channel(channelName, channel.getDescription(), channel.getVendor(), channel.getRepositories(), channel.getManifestCoordinate(), channel.getBlocklistCoordinate(), channel.getNoStreamStrategy()));
+            if (channels.size() > 1) {
+                throw new RuntimeException("More than one channel found in the list.");
+            }
+        } else if (!(channelGroup.manifestLocation.isEmpty() || channelGroup.repositoryDefs.isEmpty())) {
+            console.println(CliMessages.MESSAGES.subscribeChannel(installationDirectory, channelName));
 
-        console.println(CliMessages.MESSAGES.subscribeChannel(installationDirectory, channelName));
-
-        ChannelManifestCoordinate manifest = ArtifactUtils.manifestCoordFromString(gavUrlOrPath);
-
-        try (TemporaryFilesManager temporaryFiles = TemporaryFilesManager.getInstance()) {
-            final List<Repository> repositories = RepositoryUtils.unzipArchives(RepositoryDefinition.from(repositoryDefs), temporaryFiles);
+            ChannelManifestCoordinate manifest = ArtifactUtils.manifestCoordFromString(channelGroup.manifestLocation);
+            try (TemporaryFilesManager temporaryFiles = TemporaryFilesManager.getInstance()) {
+                final List<Repository> repositories = RepositoryUtils.unzipArchives(RepositoryDefinition.from(channelGroup.repositoryDefs), temporaryFiles);
+                channels = List.of(new Channel(channelName, null, null, repositories, manifest, null, null));
 
-            Channel channel = new Channel(channelName, null, null, repositories, manifest, null, null);
-            try (MetadataAction metadataAction = actionFactory.metadataActions(installationDirectory)) {
+            }
+        } else {
+            return ReturnCodes.INVALID_ARGUMENTS;
+        }
+        try (MetadataAction metadataAction = actionFactory.metadataActions(installationDirectory)) {
+            for (Channel channel : channels) {
```

Comment:
Since it's only possible to add one channel from file ( line #77), you don't need the loop here. There will always be just a single channel